akka-actor/src/main/scala/akka/io/SelectionHandler.scala

```diff
@@ -104,8 +104,11 @@ private[io] object SelectionHandler {
     override def supervisorStrategy = connectionSupervisorStrategy
 
     val selectorPool = context.actorOf(
-      props =
-        RandomPool(nrOfSelectors).props(Props(classOf[SelectionHandler], selectorSettings)).withDeploy(Deploy.local),
+      props = RandomPool(nrOfSelectors)
+        .withDispatcher(context.props.dispatcher)
```

Comment:
Now it has two dispatchers. Why is it needed because child actors don't inherit dispatcher configs?

Reply:
I know it looks a bit strange. The second is for the routees. The first is for the router "head" as described in https://doc.akka.io/docs/akka/current/routing.html#configuring-dispatchers

I have changed to somewhat more verbose, but clear:
```
    val selectorPool: ActorRef = {
      val routeeProps = Props(classOf[SelectionHandler], selectorSettings)
        .withDispatcher(context.props.dispatcher)
        .withDeploy(Deploy.local)
      context.actorOf(
        props = RandomPool(nrOfSelectors, routerDispatcher = context.props.dispatcher)
          .props(routeeProps)
          .withDeploy(Deploy.local),
        name = "selectors")
    }
```

no actor is inheriting the dispatcher from parent, that must be done explicitly using `.withDispatcher(context.props.dispatcher)`  otherwise it will be default-dispatcher

Reply:
Thanks for the clarification!